### ISSUE TITLE
Terminal on start-up dumps Tmux escape-color codes

### ISSUE THREAD

**Issue by tijko** [microsoft/vscode#207545]
- VS Code Version: 
1.87.1
1e790d77f81672c49be070e04474901747115651
x64

- OS Version: 
Linux tijko 6.7.9-arch1-1 # 1 SMP PREEMPT_DYNAMIC Fri, 08 Mar 2024 01:59:01 +0000 x86_64 GNU/Linux

- SysInfo:
CPUs Intel(R) Core(TM) i5 CPU M 460 @ 2.53GHz (4 x 2793)
GPU Status 2d_canvas: enabled canvas_oop_rasterization: disabled_off direct_rendering_display_compositor: disabled_off_ok gpu_compositing: enabled multiple_raster_threads: enabled_on opengl: enabled_on rasterization: enabled raw_draw: disabled_off_ok skia_graphite: disabled_off video_decode: enabled video_encode: disabled_software vulkan: disabled_off webgl: enabled webgl2: enabled webgpu: disabled_off
Load (avg) 0, 1, 1
Memory (System) 7.56GB (4.78GB free)
Process Argv	
Screen Reader no
VM	0%

- Extensions:
neo4j-vscode ada 0.1.8
ng-template	Ang 17.2.2
vim-gitpod gen 1.21.10
vscode-github-actions git	0.26.2
terraform    has 2.29.5
vscode-docker ms-1.29.0
vscode-kubernetes-tools	ms-	1.3.15
sqltools	mtx 0.28.1
sqltools-driver-mysql	mtx 0.5.1
vscode-yaml	red	1.14.0
rust rus	0.7.8
vscode-maven	vsc	0.44.0
viml Xad	2.1.2

- Environment/Shell/Term:
dmenu - 5.2 
dwm - 6.4
st - 0.9
tmux - 3.4
bash - 5.2.26(1)-release (x86_64-pc-linux-gnu)

Steps to Reproduce:

Preface - I use dmenu in tandem with dwm for a window manager.  In addition I use st as a term-emulator and tmux for term-server (multiplexer)

1.  Invoke VSCode from dmenu
2.  On start-up the Terminal dumps "10;rgb:cccc/cccc/cccc11;rgb:1818/1818/1818"

![with-tmux](https://github.com/microsoft/vscode/assets/1518884/7714569b-f8aa-4b23-922e-8fcaeac3bfa5)

This second screenshot is after I made some edits to my `.tmux.conf` file but I am not sure of whether this is a VSCode issue or not.

![with-out-tmux](https://github.com/microsoft/vscode/assets/1518884/d0ee997e-33f9-483a-bfcd-b12fcc3d8672)

I've been able to track it down to a couple of settings in my `.tmux.conf` file but I genuinely need some guidance here.

**Comment by meganrogge** [comment]
> I've been able to track it down to a couple of settings in my .tmux.conf file but I genuinely need some guidance here.

This sounds like a tmux config problem, which is not something I can help with unfortunately.

**Comment by tijko** [comment]
@meganrogge fair enough but I could have explained myself better.

The `.tmux.conf` configuration is not "incorrect" this is a perfectly functioning setup with other terminals.  It is _only_ in VSCode-Terminal where this happens.

> I've been able to track it down to a couple of settings in my .tmux.conf file ....

This was just to reflect I found out which setting that Terminal is unable to handle correctly.  I have found others that have similar problems.  So again, it isn't an error or misconfiguration per se but an issue where Terminal is able to handle something coming from TMux

**Comment by varunbpatil** [comment]
Have the same issue here. Other terminals works perfectly fine. Only vscode integrated terminal doesn't. Also, this happens only with tmux 3.4 and not 3.3a, for example.

**Comment by andersonba** [comment]
Same hre!

**Comment by tijko** [comment]
@varunbpatil I think on face value it sounded like my `.tmux.conf` file was the stem issue.

Explaining that by making some changes to the configuration file I am able to get different results.  I can post it here if it would help.

I've found a couple of other threads that bring up `tmux` having problems within `VSCode`.

**Comment by rekmarks** [comment]
I just managed to resolve this exact issue on my system.

I found that the offending setting in my `.tmux.conf` was `set -sg escape-time 0`. According to [the `tmux` man page](https://man.openbsd.org/OpenBSD-current/man1/tmux.1#escape-time), this setting determines how long `tmux` "waits [in milliseconds] after an escape is input to determine if it is part of a function or meta key sequences." Increasing the value to `100` did the trick for me.
